File: app/src/main/java/com/daniel/mobilepauker2/model/MPTextView.java
package com.daniel.mobilepauker2.model;

import android.content.Context;
import androidx.annotation.Nullable;
import android.util.AttributeSet;

import com.daniel.mobilepauker2.model.pauker_native.CardSide;
import com.daniel.mobilepauker2.model.pauker_native.Font;

public class MPTextView extends androidx.appcompat.widget.AppCompatTextView {

    public MPTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public MPTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MPTextView(Context context) {
        super(context);
    }

    public void setCard(CardSide cardside) {
        setText(cardside.getText());

        setFont(cardside.getFont());
    }

    public void setFont(@Nullable Font font) {
        ModelManager.instance().setFont(font, this);
    }
}

File: app/src/main/java/com/daniel/mobilepauker2/model/MPTextView.kt
package com.daniel.mobilepauker2.model

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.daniel.mobilepauker2.model.pauker_native.CardSide
import com.daniel.mobilepauker2.model.pauker_native.Font

class MPTextView : AppCompatTextView {
    constructor(
        context: Context?,
        attrs: AttributeSet?,
        defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr) {
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context,
        attrs
    ) {
    }

    constructor(context: Context?) : super(context) {}

    fun setCard(cardside: CardSide?) {
        text = cardside?.text
        setFont(cardside?.font)
    }

    fun setFont(font: Font?) {
        ModelManager.instance().setFont(font, this)
    }
}